mantle/kola/tests/coretest/core.go

```diff
@@ -378,3 +383,86 @@ func TestMachineID() error {
 	}
 	return nil
 }
+
+// TestRHCOSGrowpart tests whether rhcos-growpart was run successfully by checking
+// whether rhcos-growpart.service exited normally and root partition is >= 8 GB.
+func TestRHCOSGrowpart() error {
+	// check that rhcos-growpart.service was run and exited normally
+	execMainStatus, err := systemctlShow("ExecMainStatus", "rhcos-growpart.service")
+	if err != nil {
+		return err
+	}
+	if execMainStatus != "0" {
+		return fmt.Errorf("rhcos-growpart.service did not have exit code 0\n ExecMainStatus: %s", execMainStatus)
+	}
+	execMainPID, err := systemctlShow("ExecMainPID", "rhcos-growpart.service")
+	if err != nil {
+		return err
+	}
+	if execMainPID == "0" {
+		return fmt.Errorf("rhcos-growpart.service was not found")
+	}
+
+	// check that root partition is >= 8 GB
+	c := exec.Command("df", "/sysroot")
+	out, err := c.Output()
+	if err != nil {
+		return fmt.Errorf("Err: %s\n Out: %v", err, out)
+	}
+	dfOutput := strings.Split(string(out), "\n")
+	// verify header is as expected
+	if len(dfOutput) < 2 {
+		return fmt.Errorf("Expected at least two lines of df output: %q", out)
+	}
+	headerRegex := `Filesystem\s+1K-blocks\s+Used\s+Available\s+Use%\s+Mounted on`
+	if !regexp.MustCompile(headerRegex).MatchString(dfOutput[0]) {
+		return fmt.Errorf("df output has changed format: %q", dfOutput[0])
+	}
+	properties := strings.Fields(dfOutput[1])
+	if len(properties) != 6 {
+		return fmt.Errorf("Unexpected number of properties in df output: %q", dfOutput[0])
+	}
+	numBlocks, err := strconv.Atoi(properties[1])
+	if err != nil {
+		return fmt.Errorf("Error converting root partition size from string to int: %s", err)
+	}
```

Comment:
Another approach which doesn't require parsing human output is (in bash code):

```
$ srcblkdev=$(findmnt -nvr -o SOURCE /sysroot)
$ dev=$(basename "$(realpath "$srcblkdev")")
$ size=$(cat /sys/block/$dev/size)
```

Reply:
much less prone to error, thanks!